components/local-app/BUILD.yaml

```diff
@@ -42,13 +48,26 @@ packages:
         - go.sum
         - "**/*.go"
       deps:
+        - :version
         - components/supervisor-api/go:lib
         - components/gitpod-protocol/go:lib
       env:
         - CGO_ENABLED=0
         - GOOS=windows
+      prep:
+        - ["cp", "_deps/components-local-app--version/version.go", "version.go"]
       config:
         packaging: app
+    - name: version
+      type: generic
+      argdeps:
+        - localAppVersion
+      config:
+        commands:
+          - ["sh", "-c", "echo package main >> version.go"]
```

Comment:
fyi - the embed directive in Golang v1.16 can also be used for this e.g. https://www.buymeacoffee.com/smarterfish/dynamic-version-numbers-in-go-1-16

It has the bonus of not requiring a valid .go file and also standardises the way to embed any sort of static file

Reply:
Excellent point. We're using `embed` in `docker-up` already, might as well use it here.